### DISCUSSION TITLE
TcpServer里实现心跳处理

### DISCUSSION THREAD

**Discussion by jiephe** [chenshuo/muduo#614] [General]
是否可以在TcpServer这个类的loop里开个定时器，每隔一定时间去遍历所有的TcpConnectionPtr，比对TcpConnectionPtr的lastReceiveTime_和当前的时间，如果超过心跳的时间，就删除这个连接。我看已经有lastReceiveTime_这个字段了，但是没有用起来。

**Comment by chenshuo** [comment]
看起来这是清除空闲连接，而不是实现发送心跳。

**Reply by jiephe** [reply]
实现方式难道不是一样的吗，客户端每隔时间T发送一个心跳，服务端每次接收到消息都更新lastReceiveTime_。如果lastReceiveTime_和当前的时间戳间隔超过2T了，就表示这个连接已经不可用了。难道心跳还有其他的实现方式吗
